source/base/data_out_base.cc

```diff
@@ -5421,7 +5421,7 @@ namespace DataOutBase
                     for (unsigned int c = 0; c < size; ++c)
                       {
                         const auto ind =
-                          Tensor<2, dim>::unrolled_to_component_indices(c);
+                          Tensor<2, spacedim>::unrolled_to_component_indices(c);
```

Comment:
I suppose this matches what we do for vectors?

Reply:
Actually no. I think I can write this better, please wait with merging, I will update the PR.